VideoWeb/VideoWeb/ClientApp/src/app/security/refresh-token-parameter.interceptor.spec.ts

```diff
@@ -1,104 +1,187 @@
+                expect(result.body).toBe(request.body);
+                expect(next.handle).toHaveBeenCalledTimes(1);
+            }));
+
+            it('should call next and not update body if - post not matching route', fakeAsync(() => {
+                // Arrange
+                const next = jasmine.createSpyObj<HttpHandler>('HttpHandler', ['handle']);
+                let result: HttpRequest<any>;
+                next.handle.and.callFake(req => {
+                    result = req;
+                    return of({} as HttpEvent<any>);
+                });
+                const request = new HttpRequest<any>('POST', '/', 'params1');
+
+                currentIdpSubject.next(provider);
+                flush();
+
+                // Act
+                sut.intercept(request, next).subscribe(() => {});
+                flush();
+
+                // Assert
+                expect(result.body).toBe(request.body);
+                expect(next.handle).toHaveBeenCalledTimes(1);
+            }));
+
+            it('should call next and not update body if - no scope in config', fakeAsync(() => {
+                // Arrange
+                const next = jasmine.createSpyObj<HttpHandler>('HttpHandler', ['handle']);
+                let result: HttpRequest<any>;
+                next.handle.and.callFake(req => {
+                    result = req;
+                    return of({} as HttpEvent<any>);
+                });
+                const request = new HttpRequest<any>('POST', '/oauth2/v2.0/token', 'params1');
+                getSpiedPropertyGetter(securityServiceSpy, 'configuration').and.returnValue({
+                    configuration: {
+                        scope: null
+                    }
+                } as PublicConfiguration);
+
+                currentIdpSubject.next(provider);
+                flush();
+
+                // Act
+                sut.intercept(request, next).subscribe(() => {});
+                flush();
+
+                // Assert
+                expect(result.body).toBe(request.body);
+                expect(next.handle).toHaveBeenCalledTimes(1);
+            }));
+
+            it('should call next and not update body if - no body', fakeAsync(() => {
+                // Arrange
+                const next = jasmine.createSpyObj<HttpHandler>('HttpHandler', ['handle']);
+                let result: HttpRequest<any>;
+                next.handle.and.callFake(req => {
+                    result = req;
+                    return of({} as HttpEvent<any>);
+                });
+                const request = new HttpRequest<any>('POST', '/oauth2/v2.0/token', null);
+
+                currentIdpSubject.next(provider);
+                flush();
+
+                // Act
+                sut.intercept(request, next).subscribe(() => {});
+                flush();
+
+                // Assert
+                expect(result.body).toBe(request.body);
+                expect(next.handle).toHaveBeenCalledTimes(1);
```

Comment:
All these tests that don't update the body could be grouped together under a describe and have an afterEach that runs the last 6 lines of code as well as a beforeEach for some of the common setup.

Reply:
Low priority for change. Just something to be aware of.

Reply:
Leaving as is.